docs/basic-component.md

```diff
@@ -0,0 +1,261 @@
+
+```ts
+// index.ts
+export * from "./PancakeStack"
+export * from "./Pancake"
+export * from "./hooks"
+```
+
+### Component.tsx
+
+Component files should be named to match the component name (eg. the Pancake component will be named `Pancake.tsx`).
+
+See [template](#componenttsx-template) for component composition.
+
+### Styles
+
+Style files should named to match the component name (eg. the Pancake component will be named `Pancake.scss`) and live in the same directory.
+
+Keep these clean by separating styles for subcomponents in their own respective files.
+
+### Tests
+
+Test files should named to match the component/function name (eg. `Pancake.spec.tsx` and `functionName.ts`) and live in the same directory.
+
+Ideally we would want to avoid snapshot tests as they do not have meaningful assertions/expectations and are very fragile (they can break with insignificant changes). We should instead write unit and functional tests.
+
+## Component.tsx template
+
+A basic component will follow this template:
+
+```tsx
+// PancakeStack.tsx
+import React, { HTMLAttributes } from "react"
+import classnames from "classnames"
+import styles from "./PancakeStack.scss"
+
+export interface PancakeStackProps extends Omit<HTMLAttributes<HTMLDivElement>, "className"> {
+  children: React.ReactNode
+  classNameOverride?: string
+  isBooleanProp: boolean
+  hasOptionalBooleanProp?: boolean
+  onCustomFunction: () => void
+}
+
+export const PancakeStack: React.VFC<PancakeStackProps> = ({
+  children,
+  classNameOverride,
+  isBooleanProp,
+  hasOptionalBooleanProp = false,
+  onCustomFunction,
+  ...props
+}) => {
+  const [hasSyrup, setHasSyrup] = useState<boolean>(false)
+
+  const handleCustomFunction = (): boolean => {
+    onCustomFunction()
+    setHasSyrup(!hasSyrup)
+    return true
+  }
+
+  return (
+    <div
+      className={classnames(styles.pancakeStack, classNameOverride, {
+        [styles.someClass]: isBooleanProp,
+      })}
+      onSomething={handleCustomFunction}
+      {...props}
+    >
+      {children}
+      {doSomething(hasOptionalBooleanProp) && <SubComponent />}
+    </div>
+  )
+}
+
+PancakeStack.displayName = "PancakeStack"
+```
+
+### Props
+
+```tsx
+import { HTMLAttributes } from "react"
+
+export interface PancakeStackProps extends Omit<HTMLAttributes<HTMLDivElement>, "className"> {
+  children: React.ReactNode
+  classNameOverride?: string
+  isBooleanProp: boolean
+  hasOptionalBooleanProp?: boolean
+  onCustomFunction: () => void
+}
+```
+
+- Create and export an `interface` for your props in the format of `{PascalComponentName}Props`
+  - You can restrict a prop's type signature by redeclaring it (eg. changing a prop from `string | undefined` to `string`)
+  - If you need to change the type signature of an extended prop, ensure you `Omit` it otherwise it will become `never` (eg. changing a prop from `string | undefined` to `number`)
+- `Omit` `"className"` as we re-add it with an alias (`classNameOverride`)
```

Comment:
Ah. I thought it was considered condescending (at least, my vague notes say that), so was aiming for these docs to be more the "moving forward we should do this for new components (or ones without it)". @christian314159 what are your thoughts (I believe you were the one who raised it)?

If we do plan to deprecate the old naming, I can certainly still add a note in here about it. Otherwise I can just change the docs as Doug suggested.

Reply:
As discussed in stand up this morning, we'll move forward with `classNameOverride` for any newer usages, but I will document that it supersedes `classNameAndIHaveSpokenToDST`